STACpopulator/implementations/CMIP6_UofT/add_CMIP6.py

```diff
@@ -146,11 +151,20 @@ def create_stac_item(self, item_name: str, item_data: MutableMapping[str, Any])
         """
         iid = self.make_cmip6_item_id(item_data["attributes"])
 
-        try:
-            item = STAC_item_from_metadata(iid, item_data, self.item_properties_model, self.item_geometry_model)
-        except pydantic_core._pydantic_core.ValidationError:
-            print(f"ERROR: ValidationError for {iid}")
-            return -1
+        # Adding some Marble network related data to the item properties.
+        # is_replica is always False when using this app since this app is only used to populate the hostnode
+        # with the data residing on that node. Replica STAC items on other nodes in the Marble network will be
+        # populated using a different app, and there the is_replica flag will be True.
```

Comment:
Not sure what is meant exactly by replica here. In the ESGF world, this has to do with who mirrors from whom. Will MARBLE include replication of datasets?

Also, more generally, if you are going to curate copies of ESGF files that you aggregate yourself, how are you going to determine whether one of the files making the aggregation has been updated, and this update needs to be propagated to your own version?

Reply:
Also, you say that replica items would be populated by another app. Why would that be? The replica is going to be a netCDF file sitting on a THREDDS server.

Reply:
Or do you mean the replica of a STAC item (just the catalog entry, not the file) ? In that case, I think we need to work out how STAC catalogs are going to be shared from one node to another.

Reply:
> Not sure what is meant exactly by replica here. In the ESGF world, this has to do with who mirrors from whom. Will MARBLE include replication of datasets?

Not, datasets will not be replicated. We don't have enough space on the network. What I meant is that we have plans to ensure that STAC entries from one node are visible in the STAC browser of another node. As an example, at some point the Ouranos STAC catalog will also have a "CMIP6-UofT" collection whose entries be (near) duplicates of the entries on UofT's STAC catalog for the "CMIP6-UofT" (all the properties will be same and crucially, access links will be same), but `marble:host_node` property will be `UofT` and `marble:is_replica` property will be `true`. Similarly, UofT will have replica entries for catalogs on Ouranos. This way, a user can query any single node on the network to get info on data across the network.

> 
> Also, more generally, if you are going to curate copies of ESGF files that you aggregate yourself, how are you going to determine whether one of the files making the aggregation has been updated, and this update needs to be propagated to your own version?

It is my understanding that when files are updated on ESGF, they are marked with a new version number, existing files are never overridden. We should follow the same approach. The aggregations will be for all files in a single version. If, for example, one of the files in say an `EC-Earth` dataset (which come in 165 individual files) are updated, then it will be up to the manager of the node to decide if he/she wants to create a new aggregation using the updated file. My approach for UofT node would be to not touch do that unless the changes are substantial. Let me know if I didn't understand your question correctly.

Reply:
> Or do you mean the replica of a STAC item (just the catalog entry, not the file) ? In that case, I think we need to work out how STAC catalogs are going to be shared from one node to another.

Yes just STAC items. We will be doing this by completing the implementation of [STACLoader](https://github.com/crim-ca/stac-populator/blob/e7fd123c40ebd404ade128cd5c3fab7e6a167a34/STACpopulator/input.py#L134C1-L134C1). This will crawl a STAC catalog, get the JSON data, modify it as needed (for example to change `marble:is_replica`) and post it to another STAC catalog.

Reply:
Something to consider for mirrors or "replica": https://github.com/stac-extensions/alternate-assets